src/fu-history.c

```diff
@@ -1346,10 +1399,58 @@ fu_history_add_security_attribute(FuHistory *self,
 		return FALSE;
 	}
 	sqlite3_bind_text(stmt, 1, security_attr_json, -1, SQLITE_STATIC);
-	sqlite3_bind_text(stmt, 2, hsi_score, -1, SQLITE_STATIC);
+	sqlite3_bind_int(stmt, 2, hsi_score);
+	sqlite3_bind_text(stmt, 3, version, -1, SQLITE_STATIC);
+	sqlite3_bind_text(stmt, 4, diff_result, -1, SQLITE_STATIC);
+
 	return fu_history_stmt_exec(self, stmt, NULL, error);
 }
 
+gboolean
+fu_history_get_last_hsi_details(FuHistory *self, guint *ret_hsi, gchar **ret_json_attr)
+{
+	gint rc;
+	g_autoptr(sqlite3_stmt) stmt = NULL;
+
+	rc = sqlite3_prepare_v2(
+	    self->db,
+	    "SELECT hsi_score,hsi_details FROM hsi_history ORDER BY timestamp DESC LIMIT 1;",
+	    -1,
+	    &stmt,
+	    NULL);
+	if (rc != SQLITE_OK) {
+		g_debug("Error on fetching HSI history: %s", sqlite3_errmsg(self->db));
+		return FALSE;
+	}
+	while (sqlite3_step(stmt) != SQLITE_DONE) {
+		int i;
+		int num_cols = sqlite3_column_count(stmt);
+		for (i = 0; i < num_cols; i++) {
+			switch (sqlite3_column_type(stmt, i)) {
+			case (SQLITE3_TEXT):
+				if (i == 1) {
+					GString *tmp_str = g_string_new(NULL);
+					g_string_append_printf(tmp_str,
+							       "%s",
+							       sqlite3_column_text(stmt, i));
+					*ret_json_attr = g_string_free(tmp_str, FALSE);
+				}
+				break;
+			case (SQLITE_INTEGER):
+				if (i == 0)
+					*ret_hsi = sqlite3_column_int(stmt, i);
+				break;
+			default:
+				g_warning("Mismatch HSI history column format");
```

Comment:
rather than a warning here, I'd say you should plumb `error` into `fu_history_get_last_hsi_details`

Reply:
I'll modify this. Thank you.